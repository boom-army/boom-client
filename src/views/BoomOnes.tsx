import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  Box,
  Button,
  Chip,
  FormGroup,
  Grid,
  IconButton,
  Link,
  List,
  Popper,
  Stack,
  TextField,
  Typography,
} from "@mui/material";
import CollectionsIcon from "@mui/icons-material/Collections";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import ShareIcon from "@mui/icons-material/Share";
import Face6Icon from "@mui/icons-material/Face6";
import { ThemeContext } from "../contexts/theme";
import { useContext, useState } from "react";
import { AuctionLabel } from "../components/Auctions/AuctionLabel";

export const BoomOnes = () => {
  const { theme } = useContext(ThemeContext);
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(anchorEl ? null : event.currentTarget);
  };

  const open = Boolean(anchorEl);
  const id = open ? "simple-popper" : undefined;
  return (
    <Grid
      container
      spacing={1}
      mt={2}
      px={2}
      sx={{ height: "calc(100vh - 100px)", overflowY: "scroll" }}
    >
      <Grid item xs={12}>
        <Box display={"flex"} justifyContent={"space-between"}>
          <Typography variant="h2" sx={{ lineHeight: 1.5 }}>
            Boom Heroes Hero #420
          </Typography>
          <Link>
            <IconButton
              aria-describedby={id}
              type="button"
              onClick={handleClick}
              sx={{
                backgroundColor: theme.accentColor,
                color: theme.primaryColor,
                "&:hover": {
                  backgroundColor: theme.accentColor,
                  opacity: 0.8,
                },
              }}
            >
              <ShareIcon />
            </IconButton>
            <Popper
              id={id}
              open={open}
              anchorEl={anchorEl}
              placement={"left-start"}
            >
              <Box
                sx={{
                  border: `1px solid ${theme.blue.light}`,
                  borderRadius: "8px",
                  p: 1,
                  mr: 1,
                  bgcolor: theme.background2,
                }}
              >
                The content of the Popper.
              </Box>
            </Popper>
          </Link>
        </Box>
        <Box display="flex" justifyContent={"space-between"} mt={2}>
          <Typography mt={2} variant="h4">
            Family: BoomOnes Collection
          </Typography>
          <Stack mt={1} spacing={1} direction="row">
            <Chip
              label="Creator"
              variant="outlined"
              icon={
                <Face6Icon
                  sx={{ "&.MuiChip-icon": { color: theme.accentColor } }}
                />
              }
            />
            <Chip
              label="Series"
              variant="outlined"
              icon={
                <CollectionsIcon
                  sx={{ "&.MuiChip-icon": { color: theme.accentColor } }}
                />
              }
            />
          </Stack>
        </Box>
      </Grid>
      <Grid item xs={12}>
        <Box sx={{ overflow: "hidden" }}>
          <img src={"/assets/420.png"} width="100%" alt="Boom Heroes #420" />
        </Box>
      </Grid>
      <Grid item xs={12}>
        <Box
          p={2}
          sx={{
            border: `1px solid ${theme.blue.light}`,
            backgroundColor: theme.blue.dark,
          }}
        >
          <Box display={"flex"} justifyContent={"space-between"}>
            <AuctionLabel
              label="Leader"
              content={
                <Chip
                  label="@harkl"
                  variant="outlined"
                  icon={
                    <Face6Icon
                      sx={{ "&.MuiChip-icon": { color: theme.accentColor } }}
                    />
                  }
                />
              }
            />
            <AuctionLabel
              label="Leading bid"
              content={<Typography variant="h3">ðŸ’¥345 ($0.34)</Typography>}
            />
            <Box>
              <AuctionLabel
                label="Time left"
                content={
                  <Typography variant="h3" display={"inline"} ml={0.5}>
                    00:46
                  </Typography>
                }
              />
            </Box>
          </Box>
          <Box pt={4} display="flex" justifyContent={"center"}>
            <FormGroup>
              <TextField
                autoFocus
                id="bid"
                inputProps={{ inputMode: "numeric", pattern: "[0-9]*" }}
                label="Bid"
                name="bid"
                required
                type="number"
                InputLabelProps={{
                  shrink: true,
                  style: { color: theme.secondaryColor },
                }}
                InputProps={{
                  style: { color: theme.secondaryColor },
                }}
                value={345}
                onChange={() => {}}
              />
              <Button
                variant="contained"
                size="large"
                sx={{ minWidth: "50%", marginTop: "1em" }}
              >
                <Typography display={"inline"}>Place </Typography>
                <Typography
                  display={"inline"}
                  sx={{ fontWeight: 600 }}
                  ml={0.5}
                >
                  ðŸ’¥346 ($32.89)
                </Typography>
                <Typography display={"inline"} ml={0.5}>
                  Bid
                </Typography>
              </Button>
            </FormGroup>
          </Box>
          <Box mt={2} display="flex" justifyContent={"center"}>
            <Link
              href="https://dex.aldrin.com/swap?base=USDC&quote=BMA"
              target={"_blank"}
            >
              <Typography variant={"body2"}>
                Don't have any ðŸ’¥BMA? Get some here.
              </Typography>
            </Link>
          </Box>
        </Box>
      </Grid>
      <Grid item xs={12}>
        <Box mt={2}>
          <Accordion expanded={true}>
            <AccordionSummary
              expandIcon={<ExpandMoreIcon />}
              aria-controls="description-content"
              id="description-header"
            >
              <Typography>Description</Typography>
            </AccordionSummary>
            <AccordionDetails>
              <Typography>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                Suspendisse malesuada lacus ex, sit amet blandit leo lobortis
                eget.
              </Typography>
            </AccordionDetails>
          </Accordion>
        </Box>
        <Box>
          <Typography variant="h6">Bid History</Typography>
          <List></List>
        </Box>
        <Box>Metadata</Box>
        <Box>Price</Box>
      </Grid>
      <Grid item xs={12}>
        Promo box
      </Grid>
    </Grid>
  );
};
