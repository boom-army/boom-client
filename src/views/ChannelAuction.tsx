import React, { useContext, useEffect, useMemo, useRef, useState } from "react";
import { Box, useTheme } from "@mui/material";
import { ChannelStatus } from "../constants";
import { CustomResponse } from "../components/CustomResponse";
import { MeepFeed } from "../components/MeepFeed";
import { NewMessage } from "../components/Message/NewMessage";

import { atom } from "recoil";
import {
  useAddChannelMutation,
  useGetChannelByIdQuery,
} from "../generated/graphql";
import { useAnchorWallet } from "@solana/wallet-adapter-react";

export const ChannelAuction: React.FC = () => {
  const [channelId, setChannelId] = useState<string>("");
  const anchorWallet = useAnchorWallet();
  const theme = useTheme();
  const scrollRef = useRef<HTMLDivElement>();
  const parentTweetState = atom({
    key: "parentTweetState",
    default: "",
  });

  const [addChannelMutation] = useAddChannelMutation();
  const { loading, error, data, fetchMore, refetch } = useGetChannelByIdQuery({
    variables: {
      channelId,
      offset: 0,
      limit: 10,
    },
    fetchPolicy: "network-only",
    pollInterval: 10000,
  });

  useMemo(() => {
    (async () => {
      const variables = {
        mintAuthority: "harkLSUe2Puud2TVQUhHW4vs45mF1YMLU3PThPCuWd8",
        name: "BoomOnes",
        family: "AuctionHouse",
        image:
          "https://shdw-drive.genesysgo.net/F71wchw5NCG2f7jZUE6joXLJQUXU1wZdNhSN33Ww7T7K/boom-moon.png",
        description:
          "The BoomOnes Auction House. A perfect place to buy and sell your NFTs.",
        status: ChannelStatus.ACTIVE,
        channelParentId: null,
      };
      try {
        const auctionChannel = await addChannelMutation({
          variables,
        });
        setChannelId(auctionChannel.data?.addChannel?.id as string);
      } catch (error) {
        return Promise.resolve();
      }
    })();
  }, []);

  useEffect(() => {
    refetch({
      channelId: channelId as string,
      offset: 0,
    });
    scrollRef?.current?.scrollIntoView();
  }, [channelId]);

  return (
    <>
      <MeepFeed
        loading={loading}
        error={error}
        data={data?.getChannelById}
        fetchMore={fetchMore}
        parentTweetState={parentTweetState}
        scrollRef={scrollRef}
      />
      {!anchorWallet?.publicKey && (
        <Box
          sx={{
            border: `1px solid ${theme.palette.secondary}`,
            backgroundColor: theme.background2,
          }}
        >
          <CustomResponse text={"Connect your Solana wallet to chat."} />
        </Box>
      )}
      {anchorWallet?.publicKey && (
        <NewMessage
          feed={data?.getChannelById}
          channel={channelId}
          parentTweetState={parentTweetState}
          scrollRef={scrollRef}
        />
      )}
    </>
  );
};
