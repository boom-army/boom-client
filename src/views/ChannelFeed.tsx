import React, { useContext, useEffect, useRef, useState } from "react";
import { MeepFeed } from "../components/MeepFeed";
import { NewMessage } from "../components/Message/NewMessage";
import {
  TypingSubscription,
  useGetChannelByIdQuery,
  useTypingSubscription,
  useUpdateTypingStatusMutation,
} from "../generated/graphql";
import { useParams } from "react-router-dom";
import { useChannelData } from "../hooks/useChannelData";
import {
  Box,
  Button,
  Divider,
  IconButton,
  Stack,
  Typography,
  alpha,
  debounce,
  useTheme,
} from "@mui/material";
import { useWallet, useConnection } from "@solana/wallet-adapter-react";
import { userOwnsNFT } from "../utils/nfts";
import { UserContext } from "../contexts/user";
import { TypingDots } from "../components/TypingDots";
import { BOOM_CHANNEL_ID, BOOM_COLLECTION_MINT_PUBLIC_KEY } from "../utils/ids";
import { Twitter } from "@mui/icons-material";
import { CollectionStats } from "../components/Channel/CollectionStats";

function formatUserHandles(users: TypingSubscription["typing"]) {
  if (users?.length === 1) {
    return `@${users[0].user?.handle}`;
  } else {
    const handles = users?.map((user) => `@${user.user?.handle}`);
    const lastHandle = handles?.pop();
    return handles?.join(", ") + " and " + lastHandle;
  }
}

export const ChannelFeed: React.FC = () => {
  const [validNFT, setValidNFT] = useState(false);
  const { channelId } = useParams();
  const { publicKey } = useWallet();
  const { connection } = useConnection();
  const scrollRef = useRef<HTMLDivElement>();
  const { user } = useContext(UserContext);
  const theme = useTheme();
  useChannelData();

  const [updateTypingStatusMutation] = useUpdateTypingStatusMutation();
  const { data: typingdata } = useTypingSubscription({
    variables: { channelId: BOOM_CHANNEL_ID },
  });
  let typingTimeout: any;
  const handleTyping = () => {
    const userId = user?.id;
    if (userId) {
      clearTimeout(typingTimeout);
      updateTypingStatusMutation({
        variables: { channelId: BOOM_CHANNEL_ID, isTyping: true },
      });

      // User stopped typing
      typingTimeout = setTimeout(() => {
        updateTypingStatusMutation({
          variables: { channelId: BOOM_CHANNEL_ID, isTyping: false },
        });
      }, 10000);
    }
  };
  const debouncedTypingHandler = debounce(handleTyping, 500);

  useEffect(() => {
    (async () => {
      if (publicKey) {
        const ownsNft = await userOwnsNFT(
          publicKey?.toBase58(),
          BOOM_COLLECTION_MINT_PUBLIC_KEY,
          connection
        );
        setValidNFT(ownsNft);
        console.log("ownsNft", ownsNft);
      }
    })();
  }, [publicKey, connection]);

  const { loading, error, data, fetchMore, refetch } = useGetChannelByIdQuery({
    variables: {
      channelId: channelId as string,
      offset: 0,
      limit: 10,
    },
    fetchPolicy: "network-only",
    pollInterval: 10000,
  });

  useEffect(() => {
    refetch({
      channelId: channelId as string,
      offset: 0,
    });
    scrollRef?.current?.scrollIntoView();
  }, [channelId]);

  return validNFT ? (
    <Box>
      <MeepFeed
        loading={loading}
        error={error}
        data={data?.getChannelById}
        fetchMore={fetchMore}
        scrollRef={scrollRef}
      />
      <Box sx={{ position: "relative" }}>
        {typingdata?.typing && typingdata?.typing.length ? (
          <Stack
            p={1}
            sx={{
              backgroundColor: theme.blue.darkest,
              position: "absolute",
              top: -36,
              width: "100%",
            }}
            direction="row"
            spacing={0.5}
          >
            <Typography display="inline" variant="body2">
              {formatUserHandles(typingdata.typing)} is typing <TypingDots />
            </Typography>
          </Stack>
        ) : null}
        <NewMessage
          channel={channelId}
          scrollRef={scrollRef}
          typingHandler={debouncedTypingHandler}
        />
      </Box>
    </Box>
  ) : (
    <Box>
      <Typography variant="h2" p={2} textAlign="center">
        Welcome to the Boom Heroes DAO
      </Typography>
      <Box maxWidth="100%" sx={{ position: "relative" }}>
        <img
          src="https://arweave.net/gZSfTkhEe7pbu5Cnp2W9EkO6wk3bVmFYkjIlJBoqy5M?ext=png"
          width="100%"
        />
        <Box
          sx={{
            position: "absolute",
            bottom: 0,
            left: 0,
            width: "100%",
            padding: "1rem",
            backgroundColor: "rgba(0, 0, 0, 0.7)",
            color: "white",
          }}
        >
          <Typography variant="body2" sx={{ marginBottom: "1rem" }}>
            <strong>Description:</strong> The #BoomDAODAO is the first Solana
            native social DAO where everything happens on a single site directly
            on-chain.
          </Typography>
          <Typography variant="body2" sx={{ marginBottom: "1rem" }}>
            <strong>Goals:</strong> This DAOs primary focus is to design the the
            way forward for Boom DAOs in the Solana ecosystem, and to build a
            community of like-minded individuals who are passionate about the
            future of Solana.
          </Typography>
          <Typography variant="body2" sx={{ marginBottom: "1rem" }}>
            <strong>How to join:</strong> Connect your wallet containing a Boom
            Hero and you will be able to access all DAO functionality on this
            page.
          </Typography>
          <CollectionStats />
          <Divider sx={{ pt: "1rem", mb: "1rem" }} />
          <Box
            display="flex"
            justifyContent="center"
            sx={{ marginBottom: "1rem" }}
          >
            <Button
              variant="contained"
              href="https://www.tensor.trade/trade/boomheroes"
              target="_blank"
            >
              Get a Boom Hero
            </Button>
          </Box>
        </Box>
      </Box>
    </Box>
  );
};
