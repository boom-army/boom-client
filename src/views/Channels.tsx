import { Box } from "@mui/material";
import React from "react";

export const ChannelView: React.FC = () => {
  return (
    <>
      <Box>Boom</Box>
    </>
  );
};
