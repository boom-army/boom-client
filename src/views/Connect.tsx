import React from "react";
import { Connect } from "../components/Connect";

export const ConnectView = () => {
  return <Connect />;
};
