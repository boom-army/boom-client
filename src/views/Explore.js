import React from "react";
import SearchInput from "../components/Search/SearchInput";
import { styled } from "@mui/material/styles";

const Wrapper = styled("div")({
  paddingBottom: "5rem",
});

export const Explore = () => {
  return (
    <Wrapper>
      <SearchInput />
    </Wrapper>
  );
};
