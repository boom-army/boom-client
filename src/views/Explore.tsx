import React from "react";
import SearchInput from "../components/Search/SearchInput";
import { Box } from "@mui/material";

export const Explore = () => {
  return (
    <Box pb={5}>
      <SearchInput />
    </Box>
  );
};
