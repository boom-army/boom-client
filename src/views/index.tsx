export { BoomOnes } from "./BoomOnes";
export { ChannelFeed } from "./ChannelFeed";
export { Channels } from "./Channels";
export { PeopleView } from "./People";
export { Explore } from "./Explore";
export { Following } from "./Following";
export { Dashboard } from "./Dashboard";
export { News } from "./News";
export { HeroFeed } from "./HeroFeed";
// export { OGMint } from "./OGMint";
