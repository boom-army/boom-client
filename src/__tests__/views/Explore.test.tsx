import React from "react";
import { Explore } from "../../views/Explore";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SnackbarProvider } from "notistack";
import { createSerializer } from '@emotion/jest'

expect.addSnapshotSerializer(createSerializer(
  {
    classNameReplacer(className, index) {
      return `css-${index}`
    },
        // includeStyles: false 
  }
));


describe("<Explore/> component :", () => {
  test("display <Explore/> component ", async () => {
    let rendered = render(
      <SnackbarProvider>
        <Explore />
      </SnackbarProvider>,
      { wrapper: MemoryRouter }
    );
    expect(rendered).toMatchSnapshot();
  });
});

test("Assertion testing of <Explore/> component", () => {
  render(
    <SnackbarProvider>
      <Explore />
    </SnackbarProvider>,
    { wrapper: MemoryRouter }
  );
  expect(screen.getByRole("textbox")).toBeInTheDocument();
});
