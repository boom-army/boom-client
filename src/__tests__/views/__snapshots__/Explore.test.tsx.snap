// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Explore/> component : display <Explore/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-0 {
  padding-bottom: 5rem;
}

.css-1 {
  margin: 1rem 0;
  margin-left: 1rem;
}

.css-1 input {
  height: 40px;
  width: 70%;
  border-radius: 30px;
  font-size: 1rem;
  padding-left: 1.2rem;
}

@media screen and (max-width: 530px) {
  .css-1 input {
    font-size: 0.9rem;
  }
}

.css-2 .tabs {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  border-bottom: 2px solid undefined;
}

.css-2 span {
  cursor: pointer;
  margin-bottom: 0.4rem;
}

.css-2 span.active {
  border-bottom: 2px solid undefined;
  font-weight: 500;
}

.css-3 {
  height: 70vh;
  font-size: 1.1rem;
  text-align: justify;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 4rem;
}

@media screen and (max-width: 530px) {
  .css-3 {
    font-size: 1rem;
  }
}

<body>
    <div>
      <div
        class="css-0"
      >
        <div
          class="css-1"
        >
          <form>
            <input
              placeholder="Search by tags, tweets, people"
              type="text"
              value=""
            />
          </form>
        </div>
        <div
          class="css-2"
        >
          <div
            class="tabs"
          >
            <span
              class="active"
            >
              Tweets
            </span>
            <span
              class=""
            >
              Tags
            </span>
            <span
              class=""
            >
              Users
            </span>
          </div>
          <div
            class="css-3"
          >
            <p>
              Use the search bar to find tags, people and tweets
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .css-0 {
  padding-bottom: 5rem;
}

.css-1 {
  margin: 1rem 0;
  margin-left: 1rem;
}

.css-1 input {
  height: 40px;
  width: 70%;
  border-radius: 30px;
  font-size: 1rem;
  padding-left: 1.2rem;
}

@media screen and (max-width: 530px) {
  .css-1 input {
    font-size: 0.9rem;
  }
}

.css-2 .tabs {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  border-bottom: 2px solid undefined;
}

.css-2 span {
  cursor: pointer;
  margin-bottom: 0.4rem;
}

.css-2 span.active {
  border-bottom: 2px solid undefined;
  font-weight: 500;
}

.css-3 {
  height: 70vh;
  font-size: 1.1rem;
  text-align: justify;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 4rem;
}

@media screen and (max-width: 530px) {
  .css-3 {
    font-size: 1rem;
  }
}

<div>
    <div
      class="css-0"
    >
      <div
        class="css-1"
      >
        <form>
          <input
            placeholder="Search by tags, tweets, people"
            type="text"
            value=""
          />
        </form>
      </div>
      <div
        class="css-2"
      >
        <div
          class="tabs"
        >
          <span
            class="active"
          >
            Tweets
          </span>
          <span
            class=""
          >
            Tags
          </span>
          <span
            class=""
          >
            Users
          </span>
        </div>
        <div
          class="css-3"
        >
          <p>
            Use the search bar to find tags, people and tweets
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
